plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/PortListSorter.java

```diff
@@ -1,135 +1,264 @@
  * which accompanies this distribution, and is available at
  * http://www.eclipse.org/legal/epl-v10.html
- *
- * Contributors:
- *     Kiel University - initial API and implementation
  *******************************************************************************/
 package org.eclipse.elk.alg.layered.intermediate;
 
 import java.util.Collections;
 import java.util.Comparator;
+import java.util.List;
+import java.util.function.Function;
 
+import org.eclipse.elk.alg.layered.graph.LEdge;
 import org.eclipse.elk.alg.layered.graph.LGraph;
 import org.eclipse.elk.alg.layered.graph.LNode;
 import org.eclipse.elk.alg.layered.graph.LPort;
 import org.eclipse.elk.alg.layered.graph.Layer;
+import org.eclipse.elk.alg.layered.options.InternalProperties;
 import org.eclipse.elk.alg.layered.options.LayeredOptions;
+import org.eclipse.elk.alg.layered.options.PortSortingStrategy;
 import org.eclipse.elk.core.alg.ILayoutProcessor;
 import org.eclipse.elk.core.options.PortConstraints;
+import org.eclipse.elk.core.options.PortSide;
 import org.eclipse.elk.core.util.IElkProgressMonitor;
+import org.eclipse.elk.core.util.Pair;
 
 /**
  * Sorts the port lists of nodes with fixed port orders or fixed port positions. 
  * The node's list of ports is sorted beginning at the leftmost northern port, going clockwise.
  * This order of ports may be used during crossing minimization for calculating port ranks.
  * 
  * In case of {@link PortConstraints#FIXED_SIDE FIXED_SIDE} the ports are order by side
- * according to the order north - east - south - west.
+ * according to the order north - east - south - west (and depending on the value of the {@link PortSortingStrategy} 
+ * layout option)
  * In case of {@link PortConstraints#FIXED_ORDER FIXED_ORDER} the side and
  * {@link LayeredOptions#PORT_INDEX PORT_INDEX} are used if specified. Otherwise the order is inferred
  * from specified port positions. For {@link PortConstraints#FIXED_POS FIXED_POS} solely the position of
  * the ports determines the order.
  * 
  * <dl>
  *   <dt>Precondition:</dt><dd>a layered graph.</dd>
- *   <dt>Postcondition:</dt><dd>the port lists of nodes with fixed port orders are sorted.</dd>
- *   <dt>Slots:</dt><dd>Before phase 3. May additionally be used before phase 4 as well.</dd>
- *   <dt>Same-slot dependencies:</dt><dd>None.</dd>
+ *   <dt>Postcondition:</dt><dd>the port lists of nodes are sorted.</dd>
+ *   <dt>Slots:</dt><dd>Before phase 3.</dd>
+ *   <dt>Same-slot dependencies:</dt><dd>{@link NorthSouthPortPreprocessor}.</dd>
  * </dl>
  * 
  * @see LNode#getPorts()
- * @author cds
- * @author uru
- * @kieler.design 2012-08-10 chsch grh
- * @kieler.rating proposed yellow by msp
  */
 public final class PortListSorter implements ILayoutProcessor<LGraph> {
 
     /**
-     * A comparer for ports. Ports are sorted by side (north, east, south, west) in clockwise order,
-     * beginning at the top left corner.
+     * {@inheritDoc}
```

Comment:
I thought we voted against using `{@inheritDoc}`?

Reply:
This is just an unfortunate diff. The inheritDoc has been there before. 

I'm completely sure regarding our decision (which we probably should put into the documentation).